main.go

```diff
@@ -70,8 +72,60 @@ func startServer(ctx context.Context, d *daemon, tcpListener string) error {
 			w.WriteHeader(http.StatusInternalServerError)
 			_, _ = w.Write([]byte(err.Error()))
 		}
+	}
+
+	// Create a custom registry
+	reg := prometheus.NewRegistry()
+
+	requestsTotal := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_requests_total",
+			Help: "Total number of slow requests",
+		},
+		[]string{"code"},
+	)
+
+	requestDuration := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "http_request_duration_seconds",
+			Help:    "Duration of slow requests",
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"code"},
+	)
+
+	requestsInFlight := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "slow_requests_in_flight",
+		Help: "Number of slow requests currently being served",
 	})
 
+	// Register metrics with our custom registry
+	reg.MustRegister(requestsTotal)
+	reg.MustRegister(requestDuration)
+	reg.MustRegister(requestsInFlight)
+	// Instrument the slowHandler
+	instrumentedHandler := promhttp.InstrumentHandlerCounter(
+		requestsTotal,
+		promhttp.InstrumentHandlerDuration(
+			requestDuration,
+			promhttp.InstrumentHandlerInFlight(
+				requestsInFlight,
+				http.HandlerFunc(checkHandler),
+			),
+		),
+	)
+
+	// 1. Is the peer findable in the DHT?
+	// 2. Does the multiaddr work? If not, what's the error?
+	// 3. Is the CID in the DHT?
+	// 4. Does the peer respond that it has the given data over Bitswap?
+	http.Handle("/check", instrumentedHandler)
+	http.Handle("/debug/libp2p", promhttp.Handler())
+	http.Handle("/debug/http", promhttp.HandlerFor(
```

Comment:
Should these be hidden behind some kind of auth header to prevent operators from accidentally leaking more information to the public about inbound requests than they wanted to?

Reply:
Added baisc auth and documented..